include/utility.h

```diff
@@ -29,13 +32,13 @@ struct IpAddress {
     IpAddress(const char* host, const char* port);
 };
 
-std::string serializeIpAddress(const IpAddress& ipAddress);
-IpAddress deserializeIpAddress(const std::string& jsonString);
+utility::IpAddress* serializeIpAddressToProto(const IpAddress& ipAddress);
+IpAddress deserializeIpAddressFromProto(const utility::IpAddress& proto);
 
 typedef std::unordered_map<std::string, IpAddress> AddressTable;
 
-std::string serializeAddressTable(const AddressTable& addressTable);
-AddressTable deserializeAddressTable(const std::string& jsonString);
+utility::AddressTable* serializeAddressTable(const AddressTable& addressTable);
+AddressTable deserializeAddressTable(const utility::AddressTable at);
```

Comment:
This doesn't look right... why are the the parameter and return type the same?

Reply:
`utility::AddressTable` is different from `AddressTable`. `utility::AddressTable` is the proto object, while `AddressTable` is the class